src/plugins/safeSnap/components/HandleOutcomeUma.vue

```diff
@@ -31,25 +44,26 @@ const { quorum } = useQuorum(props);
 
 const plugin = new Plugin();
 
-const QuestionStates = {
-  error: -1,
-  noWalletConnection: 0,
-  loading: 1,
-  waitingForVoteConfirmation: 2,
-  noTransactions: 3,
-  completelyExecuted: 4,
-  waitingForProposal: 5,
-  waitingForLiveness: 6,
-  proposalApproved: 7
-};
-Object.freeze(QuestionStates);
+type QuestionState =
+  | 'error'
+  | 'no-wallet-connection'
+  | 'loading'
+  | 'waiting-for-vote-confirmation'
+  | 'no-transactions'
+  | 'completely-executed'
+  | 'waiting-for-proposal'
+  | 'waiting-for-liveness'
+  | 'proposal-approved';
+
+type Action1State = 'idle' | 'approve-bond' | 'submit-proposal';
+type Action2State = 'idle' | 'execute-proposal';
 
 const loading = ref(true);
-const actionInProgress = ref(false);
-const action2InProgress = ref(false);
+const action1State = ref<Action1State>('idle');
```

Comment:
Since we're doing more than just indicating whether an action is in progress or not, I changed this name to be more accurate about the intention.